packages/block-library/src/post-author/utils.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Generate Author-related blocks based on block attributes.
+ *
+ * @param {Object} attributes Block's attributes.
+ *
+ * @return {Object} Generated block.
+ */
+export function migrateToRecommendedBlocks( attributes ) {
+	const {
+		avatarSize,
+		byline,
+		showBio,
+		isLink,
+		linkTarget,
+		textAlign,
+		style,
+		...restAttributes
+	} = attributes;
+
+	return createBlock(
+		'core/group',
+		{
+			...restAttributes,
+			style: {
+				...style,
+				color: {
+					...style?.color,
+					// Duotone must be applied to the avatar block.
+					duotone: undefined,
+				},
+			},
+			layout: {
+				type: 'flex',
+				flexWrap: 'nowrap',
+				verticalAlignment: 'top',
+			},
+		},
+		[
+			showBio &&
+				createBlock( 'core/avatar', {
```

Comment:
Should this showBio check be down where the `core/post-author-biography` block is being created? Can we also make it take into account whether the author details actually contain a bio?

Reply:
> Should this showBio check be down where the core/post-author-biography block is being created?

That's right, and I should have checked howAvatar instead of showBio at this point 😅 I fixed by 491a15e.

Reply:
> Can we also make it take into account whether the author details actually contain a bio?

I think this block is mainly inserted into the Single Post template, for example. In that case, the block is not tied to a specific post, so it outputs the text 'Author Biography' instead.

Therefore, I personally think it is unnecessary to check whether the author actually has a bio. Also, if we need to check, I think we need to use [the `useSelect` hook](https://github.com/WordPress/gutenberg/blob/d339baf355b88e4b9a9c3434480824aa4d51f870/packages/block-library/src/post-author-biography/edit.js#L23-L37), but I'm worried about whether this hook will work properly inside `transforms.js`.

Reply:
Thanks for clarifying. It makes sense for the the block to remain flexible for template use.